spring-cloud-task-core/src/main/java/org/springframework/cloud/task/listener/TaskLifecycleListener.java

```diff
@@ -169,5 +178,30 @@ private void doTaskStart() {
 			logger.error("Multiple start events have been received.  The first one was " +
 					"recorded.");
 		}
+		invokeOnTaskStartup(taskExecution);
+	}
+
+	private void invokeOnTaskStartup(TaskExecution taskExecution){
+		if (taskExecutionListeners != null) {
+			for (TaskExecutionListener taskExecutionListener : taskExecutionListeners) {
+				taskExecutionListener.onTaskStartup(taskExecution);
+			}
+		}
+	}
+
+	private void invokeOnTaskEnd(TaskExecution taskExecution){
+		if (taskExecutionListeners != null) {
+			for (TaskExecutionListener taskExecutionListener : taskExecutionListeners) {
+				taskExecutionListener.onTaskEnd(taskExecution);
+			}
+		}
+	}
+
+	private void invokeOnTaskError(TaskExecution taskExecution){
+		if (taskExecutionListeners != null) {
+			for (TaskExecutionListener taskExecutionListener : taskExecutionListeners) {
+				taskExecutionListener.onTaskFailed(taskExecution);
+			}
+		}
```

Comment:
For all of these, we need to send a copy of the `TaskExecution` to the listener and limit what we copy back.  Otherwise the user could manipulate anything to detrimental consequences.